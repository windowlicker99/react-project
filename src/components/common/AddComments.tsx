import React, { ChangeEventHandler, FC, useState } from 'react';
import { getIn, useFormikContext } from 'formik';
import { Textarea } from '@components/common/form-controls/Textarea';
import { Btn } from '@components/common/form-controls/Btn';

interface AddCommentProps {
  name: string;
}

export const AddComments: FC<AddCommentProps> = ({ name }) => {
  const { errors, touched, setFieldValue } = useFormikContext();
  const [comment, setComment] = useState('');
  const handleChangeFieldValue: ChangeEventHandler<HTMLTextAreaElement> = (e) => {
    const {
      target: { value },
    } = e;
    setComment(value);
  };
  const handleResetComment = () => setComment('');
  const handleApplyComment = () => setFieldValue(name, comment);
  return (
    <>
      <Textarea
        placeholder="Add a message"
        error={getIn(touched, name) && getIn(errors, name)}
        value={comment}
        name={name}
        onChange={handleChangeFieldValue}
      />
      <div className="btn-group mt-30 flex-end">
        <Btn btnStatus="btn-lg-outline mr-15" btnText="Cancel" onClick={handleResetComment} />
        <Btn btnStatus="btn-lg-primary" btnText="Apply" disabledBtn={!comment} onClick={handleApplyComment} />
      </div>
    </>
  );
};
