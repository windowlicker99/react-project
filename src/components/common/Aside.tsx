import React, { FC, ReactNode } from 'react';
import classNames from 'classnames';

interface IAsideProps {
  header?: ReactNode;
  asideStatus?: string;
  footer?: ReactNode;
}

export const Aside: FC<IAsideProps> = ({ asideStatus = '', header, footer, children }) => (
  <div className={classNames('aside', asideStatus)}>
    <div className="aside-header">{header}</div>
    <div className="aside-body">{children}</div>
    <div className="aside-footer"> {footer}</div>
  </div>
);
