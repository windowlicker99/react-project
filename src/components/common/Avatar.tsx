import React, { FC } from 'react';
import AvatarImg from '@images/avatar.jpeg';

interface IAvatarProps {
  name: string;
  avatar: string;
}

export const Avatar: FC<IAvatarProps> = ({ name, avatar }) => (
  <div className="avatar-card">
    <div className="avatar">
      <img className="avatar-img" src={avatar || AvatarImg} alt="avatar" />
    </div>
    <span className="avatar-name">{name || 'John Stone'}</span>
  </div>
);
