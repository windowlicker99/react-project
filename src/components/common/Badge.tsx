import React, { FC } from 'react';

interface IBadgeProps {
  badgeType: string;
  badgeText: string;
  dataFor?: string;
}

export const Badge: FC<IBadgeProps> = ({ badgeType, badgeText, dataFor = '' }) => (
  <div className={`badge ${badgeType} `} {...(dataFor && { 'data-for': dataFor, 'data-tip': true })}>
    {badgeText}
  </div>
);
