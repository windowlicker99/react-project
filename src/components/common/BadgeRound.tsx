import React, { FC } from 'react';
import classNames from 'classnames';

interface IBadgeRoundProps {
  color: string;
  status?: string;
}

export const BadgeRound: FC<IBadgeRoundProps> = ({ color, status, children }) => (
  <div className={classNames('badge-round', color, status)} style={{ backgroundColor: color }}>
    {children}
  </div>
);
