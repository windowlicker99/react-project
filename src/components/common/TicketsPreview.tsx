import React, { FC, useEffect, useState } from 'react';
import { getVehicleTickets } from '@/utils/getTickets';

interface ITicketsPreviewProps {
  vehicleId?: string;
}

interface IVehicleTickets {
  closed: number;
  open: number;
  inProgress: number;
}

export const TicketsPreview: FC<ITicketsPreviewProps> = ({ vehicleId }) => {
  const [tickets, setTickets] = useState<IVehicleTickets>();

  const getTickets = async () => {
    const ticketsList = await getVehicleTickets(vehicleId);

    if (ticketsList) {
      setTickets(ticketsList);
    }
  };
  useEffect(() => {
    getTickets();
  }, [vehicleId]);
  return (
    <div className="tickets-preview">
      <div className="tickets-preview-header">
        <span className="tickets-preview-title">Tickets</span>
        <button type="button" className="tickets-preview-details">
          View Details
        </button>
      </div>
      <div className="tickets-preview-body">
        <div className="tickets-preview-info">
          {tickets?.open || 0}
          <span className="tickets-preview-data">Open</span>
        </div>
        <div className="tickets-preview-info">
          {tickets?.inProgress || 0}
          <span className="tickets-preview-data">In Progress</span>
        </div>
        <div className="tickets-preview-info">
          {tickets?.closed || 0}
          <span className="tickets-preview-data">Closed</span>
        </div>
      </div>
    </div>
  );
};
