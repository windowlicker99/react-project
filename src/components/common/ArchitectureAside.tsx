import React, { FC } from 'react';
import { Heading } from '@components/common/Heading';
import { Aside } from './Aside';
import { ICurrentElement } from '@/interfaces/interfaces';

interface IArchitectureAsideProps {
  asideStatus?: string;
  currentElement: ICurrentElement;
}

export const ArchitectureAside: FC<IArchitectureAsideProps> = ({ asideStatus = '', children, currentElement }) => (
  <Aside
    asideStatus={asideStatus}
    header={
      currentElement ? (
        <Heading headingText="Details" linkText="Edit details" linkIcon="icon-ellipsis-vertival" />
      ) : null
    }
  >
    {children}
  </Aside>
);
