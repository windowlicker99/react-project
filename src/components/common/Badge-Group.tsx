import React, { FC } from 'react';
import { EBadgeText } from '@interfaces/enums';
import { Badge } from './Badge';

interface IProps {
  type: string;
}
export const BadgeGroup: FC<IProps> = ({ type }) => (
  <div className="badge-group">
    <span className="badge-group-labe">Risk Level</span>
    <Badge badgeType={type} badgeText={EBadgeText[type as keyof typeof EBadgeText]} />
  </div>
);
