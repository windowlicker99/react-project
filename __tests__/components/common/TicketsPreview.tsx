import React from 'react';
import { render } from '@testing-library/react';
import { TicketsPreview } from '@components/common/TicketsPreview';

describe('TicketsPreview tests', () => {
  it('Should render', () => {
    const { getByText } = render(<TicketsPreview vehicleId="0" />);
    const text = getByText('Tickets');
    expect(text).toBeTruthy();
  });

  it('Should match snapshot', () => {
    const { asFragment } = render(<TicketsPreview vehicleId="0" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
