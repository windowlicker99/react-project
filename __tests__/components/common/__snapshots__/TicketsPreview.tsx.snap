// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketsPreview tests Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="tickets-preview"
  >
    <div
      class="tickets-preview-header"
    >
      <span
        class="tickets-preview-title"
      >
        Tickets
      </span>
      <button
        class="tickets-preview-details"
        type="button"
      >
        View Details
      </button>
    </div>
    <div
      class="tickets-preview-body"
    >
      <div
        class="tickets-preview-info"
      >
        0
        <span
          class="tickets-preview-data"
        >
          Open
        </span>
      </div>
      <div
        class="tickets-preview-info"
      >
        0
        <span
          class="tickets-preview-data"
        >
          In Progress
        </span>
      </div>
      <div
        class="tickets-preview-info"
      >
        0
        <span
          class="tickets-preview-data"
        >
          Closed
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
